runtime/sema/check_composite_declaration.go

```diff
@@ -819,6 +843,14 @@ func (checker *Checker) declareCompositeLikeMembersAndValue(
 			nestedCompositeDeclarationVariable :=
 				checker.valueActivations.Find(identifier.Identifier)
 
+			if ast.IsResourceDestructionDefaultEvent(identifier.Identifier) {
+				// Find the default event's type declaration
+				defaultEventType :=
+					checker.typeActivations.Find(identifier.Identifier)
+				defaultEventComposite := defaultEventType.Type.(*CompositeType)
+				compositeType.DefaultDestroyEvent = defaultEventComposite
```

Comment:
Maybe check here if `compositeType.DefaultDestroyEvent` is overwritten?

Reply:
If it is, we will get a `RedeclarationError` (we have a test for this), so it doesn't really need to be checked here.

Reply:
Ah, great! Maybe leave a comment above with this explanation